### PR TITLE
Refs [TASK][35766] Handle event Food Search Screen

### PR DESCRIPTION
**Pull request by I-am-Lien**
## Related Tickets
- [#Task 35766: Handle event Food Search Screen](https://edu-redmine.sun-asterisk.vn/issues/35766)
## WHAT
- Handle event Food Search Screen
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://github.com/framgia/coding-standards/tree/master/eng/android |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/sun/excovid19/ui/food/FragmentFood.kt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+import androidx.lifecycle.Observer
+import com.sun.excovid19.R
+import com.sun.excovid19.base.BaseFragment
+import com.sun.excovid19.data.model.Food
+import com.sun.excovid19.databinding.FragmentFoodBinding
+import com.sun.excovid19.ui.adapter.FoodAdapter
+import com.sun.excovid19.utils.AppConstant
+import com.sun.excovid19.utils.showToast
+import kotlinx.android.synthetic.main.fragment_food.*
```
Comment (huytq-2422) [inline app/src/main/java/com/sun/excovid19/ui/food/FragmentFood.kt:10]: không dùng thằng này nữa, mình chuyển sang dùng ViewBinding đi ông

File: app/src/main/java/com/sun/excovid19/ui/food/FragmentFood.kt [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    override fun initData() {
+        recycleFood.adapter = foodAdapter
+        binding?.lifecycleOwner = viewLifecycleOwner
+        viewModel.foodItems.observe(viewLifecycleOwner, Observer {
+            foodAdapter.setData(it)
+        })
+    }
+
+    override fun initActions() {
+        buttonFoodSearch.setOnClickListener {
```
Comment (huytq-2422) [inline app/src/main/java/com/sun/excovid19/ui/food/FragmentFood.kt]:
```suggestion
        viewDataBinding.buttonFoodSearch.setOnClickListener {
```

File: app/src/main/java/com/sun/excovid19/ui/food/FragmentFood.kt [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    override fun setupViews() {
+    }
+
+    override fun initData() {
+        recycleFood.adapter = foodAdapter
+        binding?.lifecycleOwner = viewLifecycleOwner
+        viewModel.foodItems.observe(viewLifecycleOwner, Observer {
+            foodAdapter.setData(it)
+        })
+    }
```
Comment (huytq-2422) [inline app/src/main/java/com/sun/excovid19/ui/food/FragmentFood.kt]:
```suggestion
        viewModel.foodItems.observe(viewLifecycleOwner, Observer(foodAdapter::setData))
    }
```

Comment (huytq-2422) [reply app/src/main/java/com/sun/excovid19/ui/food/FragmentFood.kt]: mà sao không bind luôn vào recyclerview trong xml dùng data binding + live data luôn

Comment (I-am-Lien) [reply app/src/main/java/com/sun/excovid19/ui/food/FragmentFood.kt]: tại vì cái recycle này click mới có data bind luôn thì null

**Review by thonc-2135 — APPROVED** [review]

**Review by tuannm-1276 — APPROVED** [review]

**Pull request merged**